src/CMakeLists.txt

```diff
@@ -29,46 +23,48 @@ if(WIN32)
         co/detours/disasm.cpp
     )
 else()
-    add_library(co ${LIBKIND}
+    add_library(co
         ${CC_FILES}
         co/context/context.S
     )
 endif()
 
+add_library(cocoyaxi::co ALIAS co)
+
+target_include_directories(co
+    PUBLIC
+        $<BUILD_INTERFACE:${PROJECT_SOURCE_DIR}/include>
+        $<INSTALL_INTERFACE:$<INSTALL_PREFIX>/${CMAKE_INSTALL_INCLUDEDIR}>
+)
+
+set_target_properties(co PROPERTIES
+    VERSION ${PROJECT_VERSION}
+    SOVERSION ${PROJECT_VERSION_MAJOR}
+)
 
 if(WITH_LIBCURL)
-    find_package(OpenSSL 1.1.0 REQUIRED)
-    find_package(ZLIB REQUIRED)
     find_package(CURL REQUIRED)
-    if(CURL_FOUND)
-        message(STATUS "found openssl, include: ${OPENSSL_INCLUDE_DIR}, lib: ${OPENSSL_LIBRARIES}")
-        message(STATUS "found zlib ${ZLIB_VERSION_STRING}, include: ${ZLIB_INCLUDE_DIR}, lib: ${ZLIB_LIBRARIES}")
-        message(STATUS "found libcurl ${CURL_VERSION_STRING}, include: ${CURL_INCLUDE_DIR}, lib: ${CURL_LIBRARIES}")
-        target_compile_definitions(co PRIVATE HAS_OPENSSL)
-        target_compile_definitions(co PRIVATE HAS_LIBCURL)
-        target_include_directories(co PRIVATE ${CURL_INCLUDE_DIR})
-        target_include_directories(co PRIVATE ${OPENSSL_INCLUDE_DIR})
-        target_include_directories(co PRIVATE ${ZLIB_INCLUDE_DIR})
-        target_link_libraries(co PRIVATE ${CURL_LIBRARIES})
-        target_link_libraries(co PRIVATE ${ZLIB_LIBRARIES})
-        target_link_libraries(co PRIVATE ${OPENSSL_LIBRARIES})
-    endif()
-elseif(WITH_OPENSSL)
+    target_compile_definitions(co PRIVATE HAS_LIBCURL)
+    target_link_libraries(co PRIVATE CURL::libcurl)
+endif()
+if(WITH_LIBCURL OR WITH_OPENSSL)
     find_package(OpenSSL 1.1.0 REQUIRED)
-    if(OPENSSL_FOUND)
-        message(STATUS "found openssl, include: ${OPENSSL_INCLUDE_DIR}, lib: ${OPENSSL_LIBRARIES}")
-        target_compile_definitions(co PRIVATE HAS_OPENSSL)
-        target_include_directories(co PRIVATE ${OPENSSL_INCLUDE_DIR})
-        target_link_libraries(co PRIVATE ${OPENSSL_LIBRARIES})
-    endif()
+    target_compile_definitions(co PRIVATE HAS_OPENSSL)
+    target_link_libraries(co PRIVATE OpenSSL::SSL OpenSSL::Crypto)
 endif()
 
+target_compile_features(co PUBLIC cxx_std_11)
+
 if(FPIC)
-    if(NOT MSVC)
-        target_compile_options(co PRIVATE -fPIC)
-    endif()
+    set_target_properties(co PROPERTIES POSITION_INDEPENDENT_CODE ON)
 endif()
 
+find_package(Threads REQUIRED)
+target_link_libraries(co PUBLIC Threads::Threads)
+
+if(UNIX AND NOT APPLE)
+    target_link_libraries(co PRIVATE dl)
+endif()
```

Comment:
Are you sure that dl is a direct dependency of co?

Reply:
Yes, co depends on `dlsym()` on unix.

Reply:
Indeed 👍